File: doc-examples/jdbc-vector-example-java/src/main/java/example/DocumentEmbeddingService.java
package example;

import io.micronaut.data.model.vector.Vector;
import jakarta.inject.Singleton;

@Singleton
public class DocumentEmbeddingService {

    private final DocumentEmbeddingRepository repository;

    public DocumentEmbeddingService(DocumentEmbeddingRepository repository) {
        this.repository = repository;
    }

    public void saveOne() {
        // tag::create_vector[]
        Vector vec = Vector.of(0.1d, 0.2d, 0.3d);
        repository.save(vec);
        // end::create_vector[]
    }
}

File: doc-examples/jdbc-vector-example-kotlin/src/main/kotlin/example/DocumentEmbeddingService.kt
package example

import io.micronaut.data.model.vector.Vector
import jakarta.inject.Singleton

@Singleton
class DocumentEmbeddingService(
    private val repository: DocumentEmbeddingRepository
) {
    fun saveOne() {
        // tag::create_vector[]
        val vec: Vector = Vector.of(0.1, 0.2, 0.3)
        repository.save(vec)
        // end::create_vector[]
    }
}
